enhancements/proposals/1978-scylladbdatacenter-api/README.md

```diff
@@ -0,0 +1,761 @@
+	// +optional
+	PodAntiAffinity *corev1.PodAntiAffinity `json:"podAntiAffinity,omitempty"`
+
+	// tolerations allow the pod to tolerate any taint that matches the triple <key,value,effect>
+	// using the matching operator.
+	// +optional
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+}
+
+// RackStatus is the status of a Scylla Rack
+type RackStatus struct {
+	// name is the name of datacenter this status describes.
+	Name string `json:"name,omitempty"`
+
+	// version is the current version of Scylla in use.
+	CurrentVersion string `json:"currentVersion"`
+
+	// updatedVersion is the updated version of Scylla.
+	UpdatedVersion string `json:"updatedVersion"`
+
+	// nodes is the total number of nodes requested in rack.
+	// +optional
+	Nodes *int32 `json:"nodes,omitempty"`
+
+	// nodes is the total number of nodes created in rack.
+	CurrentNodes *int32 `json:"currentNodes,omitempty"`
+
+	// updatedNodes is the number of nodes matching the current spec in rack.
+	// +optional
+	UpdatedNodes *int32 `json:"updatedNodes,omitempty"`
+
+	// readyNodes is the total number of ready nodes in rack.
+	// +optional
+	ReadyNodes *int32 `json:"readyNodes,omitempty"`
+
+	// stale indicates if the current rack status is collected for a previous generation.
+	// stale should eventually become false when the appropriate controller writes a fresh status.
+	// +optional
+	Stale *bool `json:"stale,omitempty"`
+}
+
+// ScyllaDBDatacenterStatus defines the observed state of ScyllaDBDatacenter.
+type ScyllaDBDatacenterStatus struct {
+	// observedGeneration is the most recent generation observed for this ScyllaDBDatacenter. It corresponds to the
+	// ScyllaDBDatacenter's generation, which is updated on mutation by the API Server.
+	// +optional
+	ObservedGeneration *int64 `json:"observedGeneration,omitempty"`
+
+	// conditions hold conditions describing ScyllaDBDatacenter state.
+	// To determine whether a cluster rollout is finished, look for Available=True,Progressing=False,Degraded=False.
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+
+	// version is the current version of Scylla in use.
+	// +optional
+	CurrentVersion string `json:"currentVersion,omitempty"`
+
+	// updatedVersion is the updated version of Scylla.
+	// +optional
+	UpdatedVersion string `json:"updatedVersion,omitempty"`
+
+	// nodes is the total number of nodes requested in datacenter.
+	// +optional
+	Nodes *int32 `json:"nodes,omitempty"`
+
+	// nodes is the total number of nodes created in datacenter.
+	CurrentNodes *int32 `json:"currentNodes,omitempty"`
+
+	// updatedNodes is the number of nodes matching the current spec in datacenter.
+	// +optional
+	UpdatedNodes *int32 `json:"updatedNodes,omitempty"`
+
+	// readyNodes is the total number of ready nodes in datacenter.
+	// +optional
+	ReadyNodes *int32 `json:"readyNodes,omitempty"`
+
+	// rackCount is the number of ScyllaDB racks in this cluster.
+	// +optional
+	RackCount *int32 `json:"rackCount,omitempty"`
```

Comment:
what's the use case for a projection of len(racks)?

Reply:
removed